### PR TITLE
Allow process destroying by specifying string id

### PR DESCRIPTION
**Pull request by p3g4asus**
Here I implement the @JunkFood02  idea of process destroy.

### CODE REVIEW

File: library/src/main/java/com/yausername/youtubedl_android/YoutubeDL.java [hunk 1]
```diff
@@ -143,16 +147,30 @@ public VideoInfo getInfo(YoutubeDLRequest request) throws YoutubeDLException, In
...
+            return true;
+        }
+        else
+            return false;
+    }
 
+    public YoutubeDLResponse execute(YoutubeDLRequest request, @Nullable DownloadProgressCallback callback, @Nullable String processId) throws YoutubeDLException, InterruptedException {
+        assertInit();
+        if (processId != null && id2Process.containsKey(processId))
+            return null;
```
Comment (JunkFood02) [inline library/src/main/java/com/yausername/youtubedl_android/YoutubeDL.java:173]: Maybe we can throw an error here

File: library/src/main/java/com/yausername/youtubedl_android/YoutubeDL.java [hunk 2]
```diff
@@ -143,16 +147,30 @@ public VideoInfo getInfo(YoutubeDLRequest request) throws YoutubeDLException, In
...
     }
 
     public YoutubeDLResponse execute(YoutubeDLRequest request, @Nullable DownloadProgressCallback callback) throws YoutubeDLException, InterruptedException {
-        assertInit();
+        return execute(request, callback, null);
+    }
+
+    public boolean destroyProcessById(@NonNull String id) {
+        if (id2Process.containsKey(id)) {
+            id2Process.get(id).destroy();
```
Comment (JunkFood02) [inline library/src/main/java/com/yausername/youtubedl_android/YoutubeDL.java:163]: I don't know if this is necessary, maybe a `process.isAlive()` here

**Comment by xibr** [comment]
I'll take a look and test it tonight if I have some free time.

**Review by xibr — COMMENTED** [review]
It looks good, I left some comments.

We may need to add the method in the sample app or add it in the readme file something like that

To cancel a specific download
```java
YoutubeDL.getInstance().destroyProcessById( id );
```

**Comment by p3g4asus** [comment]
I made the suggested modifications and Updated both readme and example.

**Pull request merged**